### PR TITLE
Pyoneer w/ 2.0 support

### PR DESCRIPTION
**Pull request by jimfleming**
This is a rewrite of `pyoneer` to support TF 2.0.

Some things are missing but those can go in separate PRs. For example, I removed `pynr.layers.DictFeaturizer`. I don't think anybody was using it and it should really follow the TF "nest" API rather than all of the special casing for each type. Then we could pass in an arbitrary structure of layers (wrapped in `list` or `dict`) and they would appropriately map to features. This would fit in great with `tf.data.Dataset` which allows the same thing.

I'm probably the least happy with the `pynr.nn.Moments` API so any thoughts on improving that are welcome. We normalize things in so many different ways it feels redundant and incomplete at the same time. I think they should be instances of `tf.keras.layers.Layer`, rather than `tf.keras.Model`, and follow an API similar to metrics but designed to be used inside of a model.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -5,11 +5,12 @@ name = "pypi"
 
 [packages]
 gym = "*"
-tensorflow = "*"
-tensorflow-probability = "*"
+tensorflow = "==2.0.0-beta0"
```
Comment (djsaunde) [inline Pipfile]: `pip install tensorflow==2.0.0-beta1`?

File: pyoneer/activations/__init__.py [hunk 2]
```diff
@@ -2,5 +2,4 @@
 from __future__ import division
 from __future__ import print_function
 
-from pyoneer.rl.envs.batch_env_impl import BatchEnv
-from pyoneer.rl.envs.process_env_impl import ProcessEnv
+from pyoneer.activations.activations_impl import swish
```
Comment (djsaunde) [inline pyoneer/activations/__init__.py]: Is there any reason for using `activations_impl`? Just curious. You could push the definition of `swish` in the `__init__.py` that lives under the `pyoneer/activations/` directory. I'm not sure about best practices on this.

File: pyoneer/layers/layers_impl.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    def get_config(self):
+        config = {"depth": self.depth}
+        base_config = super(OneHotEncoder, self).get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+
+class AngleEncoder(tf.keras.layers.Layer):
+    """
+    Angle encoding layer. Encodes an angle as the cosine and sine of radians.
```
Comment (djsaunde) [inline pyoneer/layers/layers_impl.py]: ...of radians or degrees.

File: setup.py [hunk 4]
```diff
@@ -4,15 +4,14 @@
 
 from setuptools import setup, find_packages
 
-REQUIRED_PACKAGES = [
-    'gym',
-    'tensorflow-probability',
-]
+REQUIRED_PACKAGES = ["gym"]
```
Comment (djsaunde) [inline setup.py]: Why is `tensorflow` not a requirement?

File: pyoneer/rl/wrappers/batch_impl.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+
+
+class Batch(object):
```
Comment (djsaunde) [inline pyoneer/rl/wrappers/batch_impl.py]: Very cool!

File: pyoneer/__init__.py [hunk 6]
```diff
@@ -2,11 +2,15 @@
 from __future__ import division
 from __future__ import print_function
 
+from pyoneer import activations
```
[the remark this answers is not in the document]
Comment (velocirabbit) [reply pyoneer/__init__.py]: `__all__` doesn't import anything. It just sets the list of objects (including modules) that would be imported when someone calls `from pyoneer import *`; you'd still have to import each module into the `pyoneer` namespace like what's being done here.

File: pyoneer/activations/__init__.py (hunk 2 above)
Comment (velocirabbit) [reply pyoneer/activations/__init__.py]: You typically don't want to define anything in an `__init__.py` file as it affects the import error. For something like importing just the `swish` function, it's not going to be a big deal, but in general, the best practice is put definitions into individual submodules within a containing module and then import them into the containing module's namespace if that's where you want them.

File: README.md [hunk 7]
```diff
@@ -64,61 +74,67 @@ In general the API tries to adhere to TensorFlow 2.0's API.
...
 
 ## Installation
 
-There are a few options of installing:
+There are a few options for installation:
 
-1. Install with `pipenv`:
+1. (Recommended) Install with `pipenv`:
 
         pipenv install pyoneer
```
[the remark this answers is not in the document]
Comment (jimfleming) [reply README.md]: It can't. We should publish on pypi under some name so it's easier to adopt. I can update the docs to reflect the current status.

**Comment by jimfleming** [comment]
> By the way, why are some tests skipped?

This is due to how tests results are detected. It reads from stdout/stderr and the TF warnings are considered "test results".

File: pyoneer/__init__.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (djsaunde) [reply pyoneer/__init__.py]: I don't think it imports the modules into the namespace, those are imported with the `import pyoneer.module` or `from pyoneer import module` syntax.

Before (with `from pyoneer import ...`):

```
In [1]: import pyoneer                                                          

In [2]: who                                                                     
pyoneer	 

In [3]: from pyoneer import *                                                   

In [4]: who                                                                     
absolute_import	 activations	 debugging	 distributions	 division	 initializers	 layers	 manip	 math	 
metrics	 nn	 print_function	 pyoneer	 rl	 schedules	 variables
```

After (with `__all__ = [...]`):

```
import pyoneer                                                          

In [2]: who                                                                     
pyoneer	 

In [3]: from pyoneer import *                                                   

In [4]: who                                                                     
absolute_import	 activations	 debugging	 distributions	 division	 initializers	 layers	 manip	 math	 
metrics	 nn	 print_function	 pyoneer	 rl	 schedules	 variables
```

In both cases, the namespace includes only `pyoneer` when you issue `import pyoneer`. Notice that in the "before" case, using `from pyoneer import *`, the namespace is cluttered with other library imports (absolute_import, division, and print_function).

**Comment by velocirabbit** [comment]
Agree with your thoughts on the Moments API. Anything that isn't expected to be used alone should be a Layer rather than a Model.

File: pyoneer/__init__.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (jimfleming) [reply pyoneer/__init__.py]: It looks like both cases produce the same outputs containing the `division` import.

Regardless, the `*`-import should be considered harmful so I'm not sure its a pattern we should support.

File: pyoneer/activations/__init__.py (hunk 2 above)
Comment (jimfleming) [reply pyoneer/activations/__init__.py]: Agree with Andrew. I'd also add that the naming convention comes from TensorFlow.

File: setup.py (hunk 4 above)
Comment (jimfleming) [reply setup.py]: When you `pip install` the package you don't want to assume what kind of TensorFlow the user wants to install. This is the same pattern followed by libraries like `tensorboard` and `tensorflow_probability` so it remains compatible with CPU/GPU variants as well as not requiring an exact version match.

The `Pipfile` is just for local installation and development so it specifies the full dependencies list.

File: pyoneer/__init__.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (djsaunde) [reply pyoneer/__init__.py]: Woops, pasted the wrong output. Second code block should be:

```
import pyoneer                                                          

In [2]: who                                                                     
pyoneer	 

In [3]: from pyoneer import *                                                   

In [4]: who                                                                     
activations	 debugging	 distributions	 initializers	 layers	 manip	 math	 
metrics	 nn	 pyoneer	 rl	 schedules	 variables
```

[the remark this answers is not in the document]
Comment (djsaunde) [reply pyoneer/__init__.py]: The other issue is that, without `__all__`, statement like `from pyoneer import print_function` are valid.
